resource-aggregate/service/service.go

```diff
@@ -114,24 +113,24 @@ func NewService(ctx context.Context, config Config, logger log.Logger, eventStor
 
 	authClient := pbAS.NewAuthorizationServiceClient(asConn.GRPC())
 
-	userDevicesManager := clientAS.NewUserDevicesManager(userDevicesChanged, authClient, config.Clients.AuthServer.PullFrequency, config.Clients.AuthServer.CacheExpiration, func(err error) { log.Errorf("resource-aggregate: error occurs during receiving devices: %v", err) })
+	nats, err := natsClient.New(config.Clients.Eventbus.NATS.Config, logger)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create nats client: %w", err)
+	}
+	grpcServer.AddCloseFunc(nats.Close)
+
+	ownerCache := clientAS.NewOwnerCache("sub", config.APIs.GRPC.OwnerCacheExpiration, nats.GetConn(), authClient, func(err error) {
+		log.Errorf("ownerCache error: %w", err)
+	})
+	grpcServer.AddCloseFunc(ownerCache.Close)
+
 	requestHandler := NewRequestHandler(config, eventStore, publisher, func(ctx context.Context, owner string, deviceIDs []string) ([]string, error) {
```

Comment:
maybe we can split into 2 functions. because we are don't need to always copy all devices for each action

Reply:
it is about O(n) vs O(ln(n))